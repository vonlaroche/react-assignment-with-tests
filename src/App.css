* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: 'Roboto', sans-serif;
}

h3 {
  color: #f55;
  font-size: 1.8em;
  line-height: 1;
  font-weight: 700;
  margin-top: 10px;
  animation: slide-up 0.4s linear;
}

.listTitle{
  text-align: center;
  padding:0;
  margin:50px 0;
  font-weight: normal;
  color: rgb(255, 255, 255);
  font-size:2em;
}


hr {
  height: 2px;
  background: #dddded;
  border: 0;
  margin: 5px 0;
  animation: slide-up 0.4s linear;
}

.mainPageSection{
  display: flex;
  flex-direction: row;
  width: 100%;
  height: 80vh;
  justify-content: space-between;
}

.sideView{
  display: flex;
  flex-direction: column;
  width:30%;
  margin-left: 20px;
}


.dialogWindowView {
  display: flex;
  flex-direction: column;
  width: 50%;
  justify-content: flex-end;
  align-items: flex-end;
  height: 100vh;
  margin-right: 50px;
  padding-bottom: 30px;  
}

.quickListView{
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  width: 50%;
  height: 80vh;
  margin-top: 30px;
  border-radius: 10px 10px 20px 20px;
  box-shadow:rgba(0, 0, 0, 0.16) 0px 5px 40px;
  opacity:1;
  background-color: #242424;
  transition: all 0.3s linear; 
  visibility: visible;
}

.quickListView.hide{
  opacity:0;
  visibility: hidden;
}

::-webkit-scrollbar {
  width: 5px;
  background-color:white;
}
::-webkit-scrollbar-track {
  margin: 0;
}

::-webkit-scrollbar-thumb {
background-color: darkgrey;
outline: none;
}


.list {
  flex:1;
  width: 100%;
  padding: 15px;
  overflow-y: auto;
  overflow-x: hidden;
  background-color: white;
  border-bottom: 1px solid rgba(36, 36, 36, 0.2);
}


@keyframes slide-up {
  0% {
      opacity: 0;
      transform: translateY(20px);
  }
  100% {
      opacity: 1;
      transform: translateY(0);
  }
}

@media all and (max-width:1440px){
  .quickListView{
    width: 60%;
  }
}

@media all and (max-width:1024px){
  .quickListView{
    width: 80%;
  }
  .listTitle{
    font-size: 1.8em;
  }
}

@media all and (max-width:768px){
  .sideView{
    width: 40%;
  }
  .quickListView{
    width: 90%;
  }
  .listTitle{
    font-size: 1.7em;
  }
}

@media all and (max-width:425px){
 .mainPageSection{
   flex-direction: column;
 }
 .sideView{
  width: 90%;
  margin:0 auto;
 }
 .listTitle{
  font-size: 1.4em;
  margin:20px 0;
}
 .dialogWindowView{
  display: flex;
  width: 90%;
  height: 100%;
  position: fixed;
  bottom: 0;
 }
 .quickListView{
   height: 100%;
 }
}
