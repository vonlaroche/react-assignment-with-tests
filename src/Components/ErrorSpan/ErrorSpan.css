.errorSpan{
    color: #ce0000;
    margin-top: 2px;
    padding: 1px;
    font-size: 0.9em;
}


@media all and (max-width:768px){
    .errorSpan{
      font-size: 0.8em;
    }
}
  