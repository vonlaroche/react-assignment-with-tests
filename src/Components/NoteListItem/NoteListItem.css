li {
    list-style: none;
    border-bottom: 1px solid rgba(110, 110, 110, 0.5);
    padding-bottom: 2px;
    min-height: 30px;
    display: flex;
    justify-content: space-between;
    align-items: flex-end;
    color:#4D4D4D;
    animation: slide-up 0.5s ease; 
    margin-top: 30px;
    margin-bottom: 5px;
    word-break: break-all;
}

  
i {
  color: #011688;
  background: none;
  border:none;
  margin: 2px;
  cursor:pointer;
}

@media all and (max-width:1366px){
  li{
    font-size: 0.83em;
    min-height: 40px;
  }
}


@media all and (max-width:768px){
  li{
    font-size: 0.8em;
  }
}



