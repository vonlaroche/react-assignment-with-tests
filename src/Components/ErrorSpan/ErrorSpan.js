import React from "react";
import emojiRegex from "emoji-regex";
import "./ErrorSpan.css";

const regex = emojiRegex();

const ErrorSpan = props => {
    const err1 = "Error, note is empty";
    const err2 = "Error, number of characters is longer than a 100";
    const err3 = "Error, contains emojis";
   
    if(props.checkNote.match(regex)){
            return <span className="errorSpan">{err3}</span>;
    }
    if (props.checkNote.length <= 0) {
        return <span className="errorSpan">{err1}</span>
    }
    if (props.checkNote.length > 100) {
        return <span className="errorSpan">{err2}</span>
    }
    return ("");
}

export default ErrorSpan;