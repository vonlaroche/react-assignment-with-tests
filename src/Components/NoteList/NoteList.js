import React from "react";
import Delay from "react-delay";
import NoteListItem from "../NoteListItem/NoteListItem";

const NoteList = props => {
    return(props.notes.map((note, index) => (
        <Delay key={index} wait={props.wait}>
          <NoteListItem
            checkNote={note}
            key={index}
            delete={() => props.onDelete(index)}>
            {note}
          </NoteListItem>
        </Delay>
      )
    ))
}

export default NoteList;