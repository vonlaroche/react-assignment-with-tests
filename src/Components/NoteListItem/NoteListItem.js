import React from "react";
import "./NoteListItem.css";
import ErrorSpan from "../ErrorSpan/ErrorSpan";

const NoteListItem = props => {
    return (<>
        <li>
            <span>{props.children}</span>
            <i className="fas fa-trash-alt deleteIcon" onClick={props.delete}></i>
        </li>
        <ErrorSpan checkNote={props.checkNote}/>
    </>);
}

export default NoteListItem;