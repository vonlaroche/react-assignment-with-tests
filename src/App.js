import React, { Component } from 'react';
import Icon from "./Components/Icon/Icon";
import NoteList from "./Components/NoteList/NoteList";
import AddNoteForm from "./Components/AddNoteForm/AddNoteForm";
import './App.css';


class App extends Component {

  state = {
    notes: ["Proper Note 1",
      "Proper Note 2",
      "ðŸ˜€",
      "",
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse rhoncus ante metus, vel consectetur urna mollis ut. Fusce vel massa vestibulum, viverra sem eget, gravida lectus. Aliquam ac dui non."],
    newNoteValue: "",
    isHidden: true
  }


  handleInputChange = event => {
    this.setState({ newNoteValue: event.target.value })
  }

  handleAddNote = event => {
    event.preventDefault();
    let noteToAdd = this.state.newNoteValue;
    this.setState({ notes: this.state.notes.concat(noteToAdd), newNoteValue: "" });
  }

  handleDeleteNote = index => {
    let array = [...this.state.notes]; // make a separate copy of the array
    array.splice(index, 1);
    this.setState({ notes: array });
  }

  handleMinMax = () => {
    if (this.state.isHidden) {
      this.setState({ isHidden: false })
    }
    else {
      this.setState({ isHidden: true })
    }
  }

  render() {

    return (<div className="App">
      <section className="mainPageSection">

        <div className="sideView">
          <h3>My notes</h3>
          <hr />
          <NoteList notes={this.state.notes} wait={2000} onDelete={this.handleDeleteNote} />
        </div>

        <div className="dialogWindowView">

          <div className={this.state.isHidden ? "quickListView hide" : "quickListView"}>
            <h3 className="listTitle">Append some new notes</h3>
            <div className="list">
              <NoteList notes={this.state.notes} wait={1000} onDelete={this.handleDeleteNote} />
            </div>

            <AddNoteForm
              onSubmit={this.handleAddNote}
              value={this.state.newNoteValue}
              onChange={this.handleInputChange}
            />
          </div>
          <Icon click={this.handleMinMax}></Icon>
        </div>

      </section>
    </div>);
  }
}

export default App;
