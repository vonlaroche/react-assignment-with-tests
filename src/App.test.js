import React from 'react';
import { create } from "react-test-renderer";
import ReactDOM from 'react-dom';
import App from './App';
import NoteList from "./Components/NoteList/NoteList";
import NoteListItem from "./Components/NoteListItem/NoteListItem";
import ErrorSpan from "./Components/ErrorSpan/ErrorSpan";
import { tsExternalModuleReference, exportAllDeclaration } from '@babel/types';
import { shallow, mount } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-enzyme';

configure({ adapter: new Adapter() });

const notesToTest = {
  validNote: "Proper note",
  emptyNote: "",
  tooLongNote: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse rhoncus ante metus, vel consectetur urna mollis ut. Fusce vel massa vestibulum, viverra sem eget, gravida lectus. Aliquam ac dui non.",
  emojiNote: "ðŸ˜€ðŸ˜€ðŸ˜€"
}

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe("Note List Item", () => {
  test("Matches the snapshot", () => {
    const noteListItem = create(<NoteListItem checkNote={notesToTest.validNote} />);
    expect(noteListItem.toJSON()).toMatchSnapshot();
  })
});

test("Error span for empty note", () => {
  const errorSpan = shallow(<ErrorSpan checkNote={notesToTest.emptyNote} />);
  expect(errorSpan.text()).toEqual("Error, note is empty");
});

test("Error span for note longer than a 100 characters", () => {
  const errorSpan = shallow(<ErrorSpan checkNote={notesToTest.tooLongNote} />);
  expect(errorSpan.text()).toContain("Error, number of characters is longer");
});

test("Error span for note that contains emoji", () => {
  const errorSpan = shallow(<ErrorSpan checkNote={notesToTest.emojiNote} />);
  expect(errorSpan.text()).toContain("emojis");
});

test("Error span for proper note", () => {
  const errorSpan = shallow(<Error checkNote={notesToTest.validNote} />);
  expect(errorSpan.text()).toEqual("");
});

