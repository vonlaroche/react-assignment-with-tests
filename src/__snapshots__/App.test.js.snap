// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note List Item Matches the snapshot 1`] = `
Array [
  <li>
    <span />
    <i
      className="fas fa-trash-alt deleteIcon"
    />
  </li>,
  "",
]
`;
